lib/Minz/ThemeExtension.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+abstract class Minz_ThemeExtension extends Minz_Extension {
+	/**
+	 * Get the list of files to use while loading the theme.
+	 *
+	 * @return array
+	 */
+	abstract protected function getThemeFiles();
+
+	public function isDefault() {
+		return false;
+	}
+
+	public function init() {
+	}
+
+	public function install() {
+		$current_user = Minz_Session::param('currentUser');
+		$themes_path = THEME_PATH . "/{$current_user}";
+
+		// Create the directory for the user themes
+		if (!file_exists($themes_path) && false === mkdir($themes_path)) {
```

Comment:
What about prior to login?

Reply:
Is that writing in `./p/themes/.../` ? This is not supposed to be a writeable location for the Web server. And it does not persist (updates, Docker...). Could we try to find another approach?

Reply:
The user is empty and the theme is copied without the user part